gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -1015,21 +1052,53 @@ nns_edge_start (nns_edge_h edge_h, bool is_server)
     return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
 
+  eh->listner_ip = _get_ip_address ();
+  eh->listner_port = _get_available_port ();
+
+  /** In the case of the server, let's publish server information. */
   eh->is_server = is_server;
-  if (!is_server && 0 == eh->recv_port) {
-    eh->recv_port = _get_available_port ();
-    if (eh->recv_port <= 0) {
-      nns_edge_loge ("Failed to start edge. Cannot get available port.");
-      nns_edge_unlock (eh);
-      return NNS_EDGE_ERROR_CONNECTION_FAILURE;
+  if (eh->is_server) {
+    if (0 != g_ascii_strcasecmp (eh->topic, "TCP_DIRECT")) {
```

Comment:
discussion: topic or connection protocol
IMO connection protocol is correct way (tcp-direct or mybrid)